### DISCUSSION TITLE
Add this plugin to an ARM64 Docker image

### DISCUSSION THREAD

**Discussion by chopper-poet** [rabbitmq/rabbitmq-delayed-message-exchange#205] [General]
Hi,if i want to integrate this plugin into arm64 rabbitmq,what should i do

thank you.

**Comment by michaelklishin** [comment]
This plugin doesn't have any architecture-specific code, so "integration" would be no different from installation of any other [RabbitMQ plugin](https://rabbitmq.com/plugins.html) that does not ship with the RabbitMQ distribution.

Since there is no such thing as an "ARM64 RabbitMQ" either, I assume your question is how to add this plugin to a Docker image. Most likely you have to create your own Docker image that inherits from an existing one (e.g. the community RabbitMQ image) and copies the plugin file to the correct [plugin directory](https://rabbitmq.com/plugins.html#plugin-directories) then [preconfigures this plugin](https://rabbitmq.com/plugins.html#enabled-plugins-file).

We won't be able to provide a set of step-by-step instructions for that, so please take it from here.

**Reply by chopper-poet** [reply]
thank you very much

**Comment by phatboyg** [comment]
Sorry to dig up an older discussion, but when I applied the new version to a 3.11.1 image which builds for both architectures (amd64 and arm64), the arm64 docker build fails:

https://github.com/MassTransit/docker-rabbitmq/actions/runs/3385966171/jobs/5624758338

I see above that you stated there aren't any architecture specific code elements in the plug-in, so I'm not sure what could be the issue. 

Any thoughts?

**Reply by binarin** [reply]
I've tried it, it works on GCP t2a instance - image builds successfully, and rabbit runs with all plugins. From your logs it appears that `rabbitmq-plugins` command segfaults, but I have no idea why.

**Reply by tobybellwood** [reply]
I've had a similar issue with 3.9.24. A bit of digging, and it works with 3.9.22 - so assuming something has changed in the main rabbitmq codebase since then to break it. Of note, this issue also occurs without any plugins specified, so probably isn't related to this specific one.

**Reply by tobybellwood** [reply]
(just checked and 3.10.6 works, 3.10.7 doesn't - so it appears that there is something amiss in that aug 2nd releases) - I'll get my thoughts together and lodge an issue with the -server repo

**Reply by pauldeng** [reply]
Are you talking about cross compile docker container?
This thread could explain it: https://stackoverflow.com/questions/73954808/how-to-run-erlang-in-arm64-docker-image-on-x86-64-host

**Reply by tobybellwood** [reply]
That's what we're doing, but it *used* to work, and now doesn't (I'm assuming it's because of something to do with the linked erlang issue) - or in the build changes that happened between the two versions https://github.com/rabbitmq/rabbitmq-server/compare/v3.10.6...v3.10.7

**Reply by michaelklishin** [reply]
This seems to be something [reported for another plugin](https://github.com/gotthardp/rabbitmq-email/pull/54#issuecomment-1283746658) in the multi-arch Docker image build context. Most likely the first Erlang version with a solution is going to be 25.1, although I can't really predict that.

Erlang JIT does not support ARM64 in versions older than 25.0, so it can either be 25.1 or a 25.0.x patch release.
